### PR TITLE
Update golang and linter

### PR DESCRIPTION
**Pull request by voidlock**
Provides a few updates, namely:

- Update to golang 1.21
- Update golangci-lint to v1.55.0

Addresses all linter issues because of the both upgrades.

### CODE REVIEW

File: tlsconfig/tlsconfig.go [hunk 1]
```diff
@@ -120,13 +120,15 @@ var (
...
 	config.MinVersion = tls.VersionTLS10
+	//nolint: staticcheck
 	config.PreferServerCipherSuites = true
 }
 
 // Default modifies config with safe defaults for standard compatibility.
 func Default(config *tls.Config) {
 	config.CipherSuites = DefaultCiphers
 	config.MinVersion = tls.VersionTLS11
+	//nolint: staticcheck
```
Comment (ypaq) [inline tlsconfig/tlsconfig.go:131]: What does staticcheck complain about?

**Review by ypaq — APPROVED** [review]
LGTM

**Review by cgfuh — APPROVED** [review]
Thank you!
the PR's description says "update to golang `1.20`" though.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -9,8 +9,8 @@ jobs:
       matrix:
         target: [ test, lint ]
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v3
+    - uses: actions/checkout@v4
```
Comment (ys) [inline .github/workflows/ci.yaml:12]: Forgot to update those on the other project. Thanks for the reminder

**Pull request merged**